overlord/assertstate/assertmgr.go

```diff
@@ -215,14 +215,10 @@ func doValidateComponent(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	db := DB(st)
-	err = snapasserts.CrossCheckResource(compsup.ComponentName(), sha3_384, expectedProv, compSize, compsup.CompSideInfo, snapsup.SideInfo, modelAs, db)
+	resRev, err := snapasserts.CrossCheckResource(compsup.ComponentName(), sha3_384, expectedProv, compSize, compsup.CompSideInfo, snapsup.SideInfo, modelAs, db)
 	if err != nil {
 		return err
```

Comment:
this is already the case but we don't seem to have a test that hits here

Reply:
Added a case in 21b440ce49c1f1decf14152ae35338758b4e1a68.